gems/pending/metadata/linux/LinuxUtils.rb

```diff
@@ -141,6 +141,44 @@ def self.parse_systemctl_list(lines)
       end.compact!
     end
 
+    def self.collect_interface(interfaces, interface)
+      mac_addr = interface[:mac_address]
+      return if mac_addr.blank?
+
+      existing_interface = interfaces[mac_addr]
+      if existing_interface.blank?
+        interfaces[mac_addr] = interface
+      else
+        interface[:name] += "," + existing_interface[:name]
+        existing_interface.merge!(interface)
+      end
+    end
+
+    def self.parse_network_interface_list(lines)
```

Comment:
LinuxAdmin already has some very similar code for getting this information, but it will always run the `ip addr show <device>`  on the local machine and then parse the output as a part of filling the mac and IP fields of the NetworkInterface object.

It looks like you guys are trying to get the network information of a remote machine, correct?  To use what we have written in the LinuxAdmin gem would be difficult in that case unless we want to expose the parsing methods as a part of the public API.

Luckily this will work just fine assuming you are running your `ip a` on some flavor of Linux, it's pretty much guaranteed to be there. We are actually making the same assumption about the command's existence and output format when we use it in LinuxAdmin.